model/project_ref.go

```diff
@@ -351,7 +351,18 @@ func (projectRef *ProjectRef) Insert() error {
 }
 
 func (p *ProjectRef) Add(creator *user.DBUser) error {
-	p.Id = mgobson.NewObjectId().Hex()
+	if p.Id != "" {
```

Comment:
Do we want to check that it's in a certain format? Or is any length/combination of letter, numbers, symbols, etc, fine?